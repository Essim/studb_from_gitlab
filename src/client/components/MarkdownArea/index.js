import React, {Component} from 'react';
import marked from 'marked';
import PropTypes from 'prop-types';


export default class MarkdownArea extends Component {

    static PropTypes = {
        content: PropTypes.string.isRequired,
    };

    constructor(props) {
        super(props);
        marked.setOptions({
            gfm: true,
            tables: true,
            breaks: false,
            pedantic: false,
            sanitize: true,
            smartLists: true,
            smartypants: false,
        });
    }

    render() {
        const markdown = {__html: marked(this.props.content || '')};
        return (<div dangerouslySetInnerHTML={markdown}></div>);
    }
}
