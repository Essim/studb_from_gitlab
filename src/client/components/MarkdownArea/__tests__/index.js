import React from 'react';
import MarkdownArea from '../';
import { shallowToJson } from 'enzyme-to-json';

const mockContent = "mockStirng";

describe('MarkdownArea', () => {
    it('should render correctly', () => {
        const output = shallow(<MarkdownArea content={mockContent}/>);
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});
