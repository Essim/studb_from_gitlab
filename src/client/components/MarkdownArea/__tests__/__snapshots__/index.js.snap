// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkdownArea should render correctly 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<p>mockStirng</p>
",
    }
  }
/>
`;
